pkg/controller/operators/olm/overrides/config.go

```diff
@@ -29,12 +29,14 @@ func (o *operatorConfig) GetConfigOverrides(ownerCSV ownerutil.Owner) (envVarOve
 		return
 	}
 
-	envVarOverrides = owner.Spec.Config.Env
-	volumeOverrides = owner.Spec.Config.Volumes
-	volumeMountOverrides = owner.Spec.Config.VolumeMounts
-	tolerationOverrides = owner.Spec.Config.Tolerations
-	resourcesOverride = owner.Spec.Config.Resources
-	nodeSelectorOverride = owner.Spec.Config.NodeSelector
+	if owner.Spec.Config != nil {
+		envVarOverrides = owner.Spec.Config.Env
+		volumeOverrides = owner.Spec.Config.Volumes
+		volumeMountOverrides = owner.Spec.Config.VolumeMounts
+		tolerationOverrides = owner.Spec.Config.Tolerations
+		resourcesOverride = owner.Spec.Config.Resources
+		nodeSelectorOverride = owner.Spec.Config.NodeSelector
+	}
```

Comment:
nit: less indentation with something like

```go
if owner.Spec.Config == nil {
    // No overrides
    return
}

envVarOverrides = owner.Spec.Config.Env
volumeOverrides = owner.Spec.Config.Volumes
volumeMountOverrides = owner.Spec.Config.VolumeMounts
tolerationOverrides = owner.Spec.Config.Tolerations
resourcesOverride = owner.Spec.Config.Resources
nodeSelectorOverride = owner.Spec.Config.NodeSelector
```

Also, we probably want to check this up front to avoid unnecessary list/filter operations.

Reply:
+1 on reducing indentation.

> Also, we probably want to check this up front to avoid unnecessary list/filter operations.

In this function, we list the subscriptions, then find the subscription that owns the CSV. `owner` is this subscription, so the `owner.Spec.Config` check requires the list and filter to happen if I'm understanding this correctly. Is there something further up the call stack you're referring to?